src/web100srv.c

```diff
@@ -1527,647 +1326,364 @@ void child_process(int parent_pipe, SSL_CTX *ssl_context, int ctlsockfd) {
-    }
-
-    if ((waiting < 0) || (mclients < 0)) {
-      log_println(6,
-                  "Fault: Negative number of clents waiting=%d, mclients=%d"
-                  ", nuke them",
-                  waiting, mclients);
-      while (head_ptr != NULL) {
-        // give message to server waiting in queue
-        child_signal = SRV_QUEUE_SERVER_BUSY;
-        write(head_ptr->pipe, &child_signal, sizeof(int));
-        close(head_ptr->pipe);
-        tpid = head_ptr->pid;
-        child_sig(-1);
-        kill(tpid, SIGTERM);
-        child_sig(tpid);
-      }
-      waiting = 0;
-      mclients = 0;
-      continue;
-    }
-
-    if ((waiting == 0) && (head_ptr != NULL)) {
-      log_println(6, "Fault: Something [%d] in queue, but no waiting clients",
-                  head_ptr->pid);
-      while (head_ptr != NULL) {
-        /* send_msg(head_ptr->ctlsockfd, SRV_QUEUE, "9977", 4); */
-        child_signal = SRV_QUEUE_SERVER_BUSY;
-        write(head_ptr->pipe, &child_signal, sizeof(int));
-        close(head_ptr->pipe);
-        tpid = head_ptr->pid;
-        child_sig(-1);
-        kill(tpid, SIGTERM);
-        child_sig(tpid);
-      }
-      waiting = 0;
-      mclients = 0;
-      continue;
+  FD_ZERO(&rfd);
+  FD_SET(listenfd, &rfd);
+  FD_SET(signalfd, &rfd);
+  if (waitforever) {
+    retcode = select(max(listenfd, signalfd) + 1, &rfd, NULL, NULL, NULL);
+  } else {
+    sel_tv.tv_sec = 3;  // 3 seconds == WAIT_TIME_SRVR
+    sel_tv.tv_usec = 0;
+    retcode = select(max(listenfd, signalfd) + 1, &rfd, NULL, NULL, &sel_tv);
+  }
+  if (retcode == -1) {
+    if (errno != EINTR) {
+      log_println(0, "Error in server's select call: %d (%s)", errno,
+                  strerror(errno));
     }
+    return NULL;
+  }
+  // If we woke up due to data on the signalfd, then read the data.
+  if (FD_ISSET(signalfd, &rfd)) {
+    read(signalfd, &signal_value, sizeof(char));
+  }
+  // If no new child is trying to connect, return
+  if (!FD_ISSET(listenfd, &rfd)) return NULL;
+  // Accept the connection, initialize variables, fire up the new child.
+  ctlsockfd = 0;
+  ctlsockfd = accept(listenfd, (struct sockaddr *)&cli_addr, &clilen);
+  if (ctlsockfd <= 0) return NULL;
+  // get meta test details copied into results
+  memcpy(&meta.c_addr, &cli_addr, clilen);
+  meta.family = ((struct sockaddr *)&cli_addr)->sa_family;
+
+  memset(rmt_addr, 0, sizeof(rmt_addr));
+  // get addr details based on socket info available
+  tmp_addr = I2AddrBySockFD(get_errhandle(), ctlsockfd, False);
+  addr2a(&cli_addr, rmt_addr, sizeof(rmt_addr));
+
+  rmt_host_strlen = sizeof(rmt_host);
+  memset(rmt_host, 0, rmt_host_strlen);
+  I2AddrNodeName(tmp_addr, rmt_host, &rmt_host_strlen);
+  log_println(4, "New connection received from 0x%x [%s] sockfd=%d.", tmp_addr,
+              tmpstr, ctlsockfd);
+  procstatusenum = PROCESS_STARTED;
+  proctypeenum = CONNECT_TYPE;
+  protolog_procstatus(getpid(), getCurrentTest(), proctypeenum, procstatusenum,
```

Comment:
Can we simplify this to 
```
protolog_procstatus(getpid(), getCurrentTest(), CONNECT_TYPE, PROCESS_STARTED, ctlsockfd);
```
and just delete `procstatusenum` and `proctypeenum`?